src/pudl/transform/ferc1.py

```diff
@@ -3042,8 +3042,49 @@ def transform_main(self, df: pd.DataFrame) -> pd.DataFrame:
         """The main table-specific transformations, affecting contents not structure.
 
         Annotates and alters data based on information from the XBRL taxonomy metadata.
+
+        Fix instances where electric_plant_sold should be positive. This is not all
+        instance where electric_plant_sold is negative, rather some of the instances
+        where electric_plant_sold is negative AND the calculated value for
+        electric_plant_in_service (the field that uses electric_plant_sold as a
+        subcomponent) does not match it's calculated value.
         """
-        return super().transform_main(df).pipe(self.apply_sign_conventions)
+        df = super().transform_main(df).pipe(self.apply_sign_conventions)
+        # Create a copy of the df with the calculations so we can apply the fixes to
+        # the original df and run the calculations later (in transform_end)
+        df_calc = df.pipe(self.reconcile_table_calculations).copy()
+        index_group_1 = ["report_year", "utility_id_ferc1"]
+        index_group_2 = ["report_year", "utility_id_ferc1", "ferc_account_label"]
+        # Get the index values for electric_plant_in_service fields that don't match
+        # their calculations
+        bad_calcs_pis_index = (
+            df_calc[
+                (df_calc["ferc_account_label"] == "electric_plant_in_service")
+                & (df_calc["ending_balance"] != df_calc["calculated_amount"])
+            ]
+            .set_index(index_group_1)
+            .index
+        )
+        # Use that index to get the whole group of ferc_account_labels associated
+        # with that bad calculation
+        bad_calc_groups_df = df_calc.set_index(index_group_1).loc[bad_calcs_pis_index]
+        # Get the index values for the instances of electric_plant_sold that exist
+        # within the groups with bad calculations.
+        bad_calc_eps_index = (
+            bad_calc_groups_df[
+                bad_calc_groups_df["ferc_account_label"] == "electric_plant_sold"
+            ]
+            .reset_index()
+            .set_index(index_group_2)
+            .index
+        )
+        df = df.reset_index().set_index(index_group_2)
+        # Replace the electric_plant_sold ending_balance values identified above with
+        # their absolute value. Also enforce the schema so we can run the calcs again.
+        df.loc[bad_calc_eps_index, "ending_balance"] = abs(df["ending_balance"])
```

Comment:
Would be nice to have a logger statement here counting the # of values changes so we can tell if things go off the rails.